### DISCUSSION TITLE
interested in helping out with the docs

### DISCUSSION THREAD

**Discussion by brossetti1** [ljosberinn/personal-react-boilerplate#679] [General]
hey @ljosberinn, 

I really like how you have set this mix up with chakra and nextjs, I see you have an unrendered docs component, wondering if you need help with that bit. Id be willing to spend some time on the project if you are. If you are not then disregard, otherwise if you open some issues on steps for pushing docs forward I can pick them up when I have time. Thanks again for setting this project up!

**Comment by ljosberinn** [comment]
Thank you, I really appreciate the idea but I'm soaked with work + fairly burnt out at the moment so I don't have capacities left for this project at the current time :/ 

FWIW I do have older doc stubs here: https://github.com/ljosberinn/next-karma-docs

I recently removed the link to the `/docs` page because I felt it's better, easier and faster conveyed through a GitHub readme after all, so updating that to reflect the changes made to the repo since then would be a first step..

Most notable changes would be:
- reworked/modularized OAuth2 (including support for Battle.net as another provider example, however it's in a different repo at this time)
- dropped `react-i18next` in favor of a more `next-translate`-y approach
- using Next.js 10 i18n solution for routing, so no language detection via custom code anymore

**Comment by brossetti1** [comment]
not a problem, and I totally understandable - burnout is for real - thanks for the details! 

I can start with reformatting the markdown and adding anything Ive picked up so far from `karma`. I guess it might good to understand what your goal of this repo is. Since you are a chakra core contributor I was thinking that maybe you see `karma` as a separate package eventually that could be a heavier (or more offical) approach to the [standard way of integrating Nextjs and Chakra](https://github.com/vercel/next.js/tree/canary/examples/with-chakra-ui-typescript). 

If thats the case, then it would make sense to approach this effort and karma could eventually be pulled out of here into its own package. Is that the goal of the project?

or, like the name of the repo describes, is this more of an open source personal starter for you?

**Comment by ljosberinn** [comment]
We've only very briefly talked about integrating it into the Chakra family, but that was months ago and tbh I don't see this repo to be package-able. I've thought about splitting up into:

- @karma/i18n
- @karma/sentry
- @karma/chakra
- @karma/auth
- @karma/test-utils

and maybe some leftovers in @karma/core

and found that usually the code is too interweaved with other parts than being able to separate it properly. Also considering to actually drop the fairly deep Chakra implementation to make it even less opinionated, which might actually make it packagable then, because Chakra is arguably by far the largest package. Karma apart from it is less than 4 kB, currently _including_ the pre-built components.

So my plan would be to move it away from this repo, `personal-react-boilerplate` which technically is in v3 at this point and over a year old, to the currently empty `next-karma` repo and post it as more of a `nextjs best practices` thing here and there, contrasting _actual_ boilerplates like bison or blitz.
